mbtiles/scripts/cli.py

```diff
@@ -359,6 +326,31 @@ def mbtiles(
         # Initialize iterator over output tiles.
         tiles = mercantile.tiles(west, south, east, north, range(minzoom, maxzoom + 1))
 
+        def insert_results(cursor, tile, contents, img_ext=None, image_dump=None):
+            if contents is None:
+                log.info("Tile %r is empty and will be skipped", tile)
+                return
+
+            # MBTiles have a different origin than Mercantile/tilebelt.
+            tiley = int(math.pow(2, tile.z)) - tile.y - 1
+
+            # Optional image dump.
+            if image_dump:
+                img_name = "{}-{}-{}.{}".format(tile.x, tiley, tile.z, img_ext)
+                img_path = os.path.join(image_dump, img_name)
+                with open(img_path, "wb") as img:
+                    img.write(contents)
+
+            # Insert tile into db.
+            log.info("Inserting tile: tile=%r", tile)
+
+            cursor.execute(
+                "INSERT INTO tiles "
+                "(zoom_level, tile_column, tile_row, tile_data) "
+                "VALUES (?, ?, ?, ?);",
+                (tile.z, tile.x, tiley, sqlite3.Binary(contents)),
+            )
+
```

Comment:
This was defined outside the command before, but not tested, so I'm moving it into its parents' basement (so to speak).